packages/x-components/src/components/auto-progress-bar.vue

```diff
@@ -5,43 +5,49 @@
 </template>
 
 <script lang="ts">
-  import Vue from 'vue';
-  import { Component, Prop } from 'vue-property-decorator';
+  import { computed, defineComponent } from 'vue';
 
   /**
-   * An auto progress bar component.
+   * The auto progress bar component is useful for displaying a visual indicator of numerical data
+   * in a bar shape.
    *
    * @public
    */
-  @Component
-  export default class AutoProgressBar extends Vue {
-    /**
-     * A boolean flag indicating if the bar is loading.
-     *
-     * @public
-     */
-    @Prop({ default: true })
-    public isLoading!: boolean;
-
-    /**
-     * The duration in seconds of the progress bar.
-     *
-     * @public
-     */
-    @Prop({ default: 5 })
-    public durationInSeconds!: number;
-
-    /**
-     * Computed property to calculate the animation's duration.
-     *
-     * @returns The CSS styles of the animation.
-     *
-     * @internal
-     */
-    protected get cssStyles(): Partial<CSSStyleDeclaration> {
-      return { animationDuration: `${this.durationInSeconds}s` };
+  export default defineComponent({
+    props: {
+      /**
+       * A boolean flag indicating if the bar is loading.
+       *
+       * @public
+       */
+      isLoading: {
+        type: Boolean,
+        default: true
+      },
+      /**
+       * The duration in seconds of the progress bar.
+       *
+       * @public
+       */
+      durationInSeconds: {
+        type: Number,
+        default: 5
+      }
+    },
+    setup(props) {
+      /**
+       * Computed property to calculate the animation's duration.
+       *
+       * @returns The CSS styles of the animation.
+       *
+       * @internal
+       */
+      const cssStyles = computed(() => ({ animationDuration: `${props.durationInSeconds}s` }));
```

Comment:
We have three options to type this:

Typing the constant: 
`const cssStyles: ComputedRef<Partial<CSSStyleDeclaration>> = computed(() => ({
        animationDuration: `${props.durationInSeconds}s`
      }));`

Typing the getter: 
`const cssStyles = computed(
        (): Partial<CSSStyleDeclaration> => ({ animationDuration: `${props.durationInSeconds}s` })
      );`

Typing both (I think it's a bit redundant doing it in both, as the getter would be inferred if we type the constant, and, in fact, it should be `ComputedGetter<Partial<CSSStyleDeclaration>>` ):
`const cssStyles: ComputedRef<Partial<CSSStyleDeclaration>> = computed(
        (): Partial<CSSStyleDeclaration> => ({ animationDuration: `${props.durationInSeconds}s` })
      );`

Reply:
Reading the vue doc, they suggest to type the computed using the generic argument: [vue doc](https://vuejs.org/guide/typescript/composition-api.html#typing-computed)

```
  const cssStyles = computed<Partial<CSSStyleDeclaration>>(() => ({
        animationDuration: `${props.durationInSeconds}s`
      }));
```